src/MDMTransaction.m

```diff
@@ -30,17 +32,65 @@ - (instancetype)init {
 }
 
 - (void)addPlan:(NSObject<MDMPlan> *)plan toTarget:(id)target {
-  MDMTransactionLog *log = [MDMTransactionLog new];
-  log.plans = @[ [plan copy] ];
-  log.target = target;
+  [self commonAddPlan:plan toTarget:target withName:nil];
+}
+
+- (void)addPlan:(NSObject<MDMPlan> *)plan toTarget:(id)target withName:(NSString *)name {
+  [self removePlanNamed:name fromTarget:target];
+  [self commonAddPlan:plan toTarget:target withName:name];
+}
+
+- (void)commonAddPlan:(NSObject<MDMPlan> *)plan toTarget:(id)target withName:(NSString *)name {
+  MDMTransactionLogType transactionLogType = name == nil ? MDMTransactionLogTypeUncategorized : MDMTransactionLogTypeAddNamedPlan;
+  MDMTransactionLog *log = [[MDMTransactionLog alloc] initWithPlan:[plan copy] target:target name:name transactionLogType:transactionLogType];
   [_logs addObject:log];
 }
 
+- (void)removePlanNamed:(nonnull NSString *)name fromTarget:(nonnull id)target {
+  MDMTransactionLog *log = [[MDMTransactionLog alloc] initWithPlan:nil target:target name:name transactionLogType:MDMTransactionLogTypeRemoveNamedPlan];
+  [_logs removeObject:log];
```

Comment:
This should be `_logs addObject:log`, or our performer group will never receive the operation.